src/Dapr.AzureFunctions.Extension/Triggers/DaprTopicTriggerBindingProvider.cs

```diff
@@ -37,36 +37,49 @@ public DaprTopicTriggerBindingProvider(DaprServiceListener serviceListener, INam
                 return Utils.NullTriggerBindingTask;
             }
 
+            // Resolve names in pub/sub, topic, and route from settings
+            if (!this.nameResolver.TryResolveWholeString(attribute.Name, out var name))
+            {
+                name = attribute.Name;
+            }
+
             string topic = TriggerHelper.ResolveTriggerName(parameter, this.nameResolver, attribute.Topic);
 
-            // Verison 0.8 only support route to be the same as the topic name
+            if (attribute.Route is null || !this.nameResolver.TryResolveWholeString(attribute.Route, out var route))
+            {
+                route = attribute.Route ?? topic;
+            }
+
             return Task.FromResult<ITriggerBinding?>(
-                new DaprTopicTriggerBinding(this.serviceListener, topic, route: topic, parameter));
+                new DaprTopicTriggerBinding(this.serviceListener, name, topic, route, parameter));
```

Comment:
Added support for separating route and topic name (gap called out in the comments previously)